### PR TITLE
Update mini bot

### PR DESCRIPTION
**Pull request by ihernandez3647**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -70,8 +59,8 @@ public class RobotContainer {
     public RobotContainer() {
         this.fieldRelative = Constants.Swerve.isFieldRelative;
         this.openLoop = Constants.Swerve.isOpenLoop;
-        s_Swerve.setDefaultCommand(new TeleopSwerve(s_Swerve, vision, driver, translationAxis,
-            strafeAxis, rotationAxis, fieldRelative, openLoop));
+        s_Swerve.setDefaultCommand(new TeleopSwerve(s_Swerve, vision, driver,
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:62]: lets get rid of the vision stuff.  No longer needed

File: src/main/java/frc/robot/commands/TeleopSwerve.java [hunk 2]
```diff
@@ -19,35 +18,29 @@ public class TeleopSwerve extends CommandBase {
...
-        this.translationAxis = translationAxis;
-        this.strafeAxis = strafeAxis;
-        this.rotationAxis = rotationAxis;
         this.fieldRelative = fieldRelative;
         this.openLoop = openLoop;
     }
 
     @Override
     public void execute() {
         vision.update();
```
Comment (legoguy1000) [inline src/main/java/frc/robot/commands/TeleopSwerve.java:40]: remove vision

File: src/main/java/frc/robot/commands/TeleopSwerve.java [hunk 3]
```diff
@@ -19,35 +18,29 @@ public class TeleopSwerve extends CommandBase {
...
-    private int rotationAxis;
+    private CommandXboxController controller;
     private Vision vision;
 
     /**
      * Driver control
      */
-    public TeleopSwerve(Swerve s_Swerve, Vision vision, Joystick controller, int translationAxis,
-        int strafeAxis, int rotationAxis, boolean fieldRelative, boolean openLoop) {
+    public TeleopSwerve(Swerve s_Swerve, Vision vision, CommandXboxController controller,
```
Comment (legoguy1000) [inline src/main/java/frc/robot/commands/TeleopSwerve.java:27]: remove vision

File: src/main/java/frc/robot/commands/TeleopSwerve.java [hunk 4]
```diff
@@ -19,35 +18,29 @@ public class TeleopSwerve extends CommandBase {
     private boolean openLoop;
 
     private Swerve s_Swerve;
-    private Joystick controller;
-    private int translationAxis;
-    private int strafeAxis;
-    private int rotationAxis;
+    private CommandXboxController controller;
     private Vision vision;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/commands/TeleopSwerve.java:22]: remove vision

File: src/main/java/frc/robot/subsystems/Swerve.java [hunk 5]
```diff
@@ -31,7 +34,8 @@ public Swerve() {
         // gyro.configFactoryDefault();
         zeroGyro();
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/Swerve.java:35]: remove this

File: src/main/java/frc/robot/subsystems/Swerve.java [hunk 6]
```diff
@@ -130,6 +135,14 @@ public void zeroGyro() {
         gyro.zeroYaw();
     }
 
+    /**
+     * Resets the gyro field relative driving offset
+     */
+    public void resetFieldRelativeOffset() {
+        // gyro.zeroYaw();
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/Swerve.java:142]: delete this line

**Review by legoguy1000 — CHANGES_REQUESTED** [review]
- Delete the Autos and associated files.  Leave the "Resnick Auto" as a place holder. (lines 37-39)
- Delete the `Ultrasonic()` and associated files. (lines 56, 970
- Update the `getAutonomousCommand()` method to be like https://github.com/Frc5572/FRC2022/blob/main/src/main/java/frc/robot/RobotContainer.java#L255

File: src/main/java/frc/robot/subsystems/Swerve.java [hunk 7]
```diff
@@ -49,14 +60,20 @@ public Swerve() {
      */
     public void drive(Translation2d translation, double rotation, boolean fieldRelative,
         boolean isOpenLoop) {
+
+        // If pidTurn has value, it will override driver steering!!
+        if (pidTurn != 0) {
+            rotation = pidTurn;
+        }
+
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/Swerve.java:68]: Delete this

File: src/main/java/frc/robot/subsystems/Swerve.java [hunk 8]
```diff
@@ -143,41 +160,69 @@ public void resetFieldRelativeOffset() {
...
+        float yaw = gyro.getYaw();
+        return (Constants.Swerve.invertGyro) ? "Yaw: " + (360 - yaw) : "Yaw: " + yaw;
+    }
+
     @Override
     public void periodic() {
-        swerveOdometry.update(getYaw(), getPositions());
+        Rotation2d yaw = getYaw();
+        // swerveOdometry.update(getYaw(), getPositions());
+        swerveOdometry.update(new Rotation2d(yaw.getRadians()), getPositions());
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/Swerve.java:176]: `getYaw()` returns a `Rotation2d` so no reason to convert to radians and then back. `swerveOdometry.update(getYaw(), getPositions());` will work

File: src/main/java/frc/robot/subsystems/Swerve.java [hunk 9]
```diff
@@ -143,41 +160,69 @@ public void resetFieldRelativeOffset() {
...
+        for (SwerveModule mod : swerveMods) {
             positions[mod.moduleNumber] = mod.getPosition();
+
         }
         return positions;
     }
+
+    public void useOutput(double output) {
+        pidTurn = output * 4;
+    }
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/Swerve.java:227]: Not needed

**Review by legoguy1000 — CHANGES_REQUESTED** [review]
minor things then GTG

**Review by agrinmanriv0537 — APPROVED** [review]

**Review by legoguy1000 — APPROVED** [review]
Looks GTG

**Pull request merged**